Uninitialized local variable warning during compilation

This warning was emitted during compilation using VS2017:
```
c:\php-sdk\php\vc15\x64\php-src\ext\igbinary\src\php7\igbinary.c(2549) : warning C4700: uninitialized local variable 'ref' used
```

This warning was emitted when compiling https://github.com/igbinary/igbinary/commit/cb1b4ba111668d29bd432c045dfcf6fac0dd70f0 .

--- a/src/php5/hash_si_ptr.c
+++ b/src/php5/hash_si_ptr.c
@@ -73,8 +73,6 @@ int hash_si_ptr_init(struct hash_si_ptr *h, size_t size) {
 /* }}} */
 /* {{{ hash_si_ptr_deinit */
 void hash_si_ptr_deinit(struct hash_si_ptr *h) {
-	size_t i;
-
 	free(h->data);
 	h->data = NULL;
 

--- a/src/php7/igbinary.c
+++ b/src/php7/igbinary.c
@@ -1708,7 +1708,7 @@ static int igbinary_serialize_zval(struct igbinary_serialize_data *igsd, zval *z
 /* {{{ igbinary_unserialize_data_init */
 /** Inits igbinary_unserialize_data_init. */
 inline static int igbinary_unserialize_data_init(struct igbinary_unserialize_data *igsd) {
-	smart_string empty_str = { 0 };
+	smart_string empty_str = {0, 0, 0};
 
 	igsd->buffer = NULL;
 	igsd->buffer_end = NULL;
@@ -2556,7 +2556,8 @@ inline static int igbinary_unserialize_object(struct igbinary_unserialize_data *
 	}
 
 	{
-		struct igbinary_value_ref ref = {0};
+		// The actual value of ref is unused. We use ref_n later in this function, after creating the object.
+		struct igbinary_value_ref ref = {{0}, 0};
 		ref_n = igsd_append_ref(igsd, ref);
 		if (ref_n == SIZE_MAX) {
 			zend_string_release(class_name);

